homeassistant/components/sensor/__init__.py

```diff
@@ -246,6 +247,22 @@ def device_class(self) -> SensorDeviceClass | None:
             return self.entity_description.device_class
         return None
 
+    @final
+    @property
+    def numeric_state_expected(self) -> bool:
```

Comment:
This method is partially flawed.

For example, a sensor without state class, without a unit of measurement and without precision and without a device class will return 0.

The fact is: We don't know. Could be numeric, could be textual state 🤷

Reply:
That results in the same unknown issue.

Reply:
An sensor can have a `''` as a state, but a numeric sensor can not.

Reply:
That's why I wasn't sure about naming. I didn't like `is_numeric` because it implited the value had been checked

I proposed `numeric_state_expected` because if that condition is True, then it HAS to be a number.
If it returns False, then it CAN be a number, but it doesn't HAVE to be.

Reply:
Adjusted the docstr

Reply:
Ah the state can be `None` too, may be better to just set it to `Return true if the sensor must be numeric.`?